check-vulnerabilities/check_vulnerabilities.py

```diff
@@ -0,0 +1,348 @@
+"""
+Security vulnerabilities script.
+
+Notes
+-----
+Script for detecting vulnerabilities on a given repo and creating
+associated security vulnerability advisories.
+"""
+
+import hashlib
+import json
+import os
+import sys
+from typing import Any, Dict
+
+import click
+import github
+
+TOKEN = os.environ.get("DEPENDENCY_CHECK_TOKEN", None)
+PACKAGE = os.environ.get("DEPENDENCY_CHECK_PACKAGE_NAME", None)
+REPOSITORY = os.environ.get("DEPENDENCY_CHECK_REPOSITORY", None)
+DRY_RUN = True if os.environ.get("DEPENDENCY_CHECK_DRY_RUN", None) else False
+ERROR_IF_NEW_ADVISORY = (
+    True if os.environ.get("DEPENDENCY_CHECK_ERROR_EXIT", None) else False
+)
+CREATE_ISSUES = (
+    True if os.environ.get("DEPENDENCY_CHECK_CREATE_ISSUES", None) else False
+)
+
+
+def dict_hash(dictionary: Dict[str, Any]) -> str:
+    """MD5 hash of a dictionary."""
+    dhash = hashlib.md5()
+    # We need to sort arguments so {'a': 1, 'b': 2} is
+    # the same as {'b': 2, 'a': 1}
+    encoded = json.dumps(dictionary, sort_keys=True).encode()
+    dhash.update(encoded)
+    return dhash.hexdigest()
+
+
+def check_vulnerabilities():
+    """Check library and third-party vulnerabilities."""
+    new_advisory_detected = False
+    # Check that the needed environment variables are provided
+    if any([v is None for v in [TOKEN, REPOSITORY, PACKAGE]]):
+        raise RuntimeError(
+            "Required environment variables are not defined. Enter value for ",
+            "'DEPENDENCY_CHECK_TOKEN', 'DEPENDENCY_CHECK_PACKAGE_NAME', ",
+            "'DEPENDENCY_CHECK_REPOSITORY'.",
+        )
```

Comment:
Why not unravel this into three ifs and raise an error specific for each key? It seems more useful to know which specific key is missing from my point of view.

Reply:
Saving time mostly - I can break it down nonetheless